client/state/selectors/get-current-query-arguments.ts

```diff
@@ -17,9 +18,7 @@ import type { IAppState } from 'calypso/state/types';
  * Kept for backwards compatibility in case any null checks.
  * Investigate and refactor.
  */
-export function getCurrentQueryArguments(
-	state: IAppState
-): IAppState[ 'route' ][ 'query' ][ 'current' ] | null {
+export function getCurrentQueryArguments( state: IAppState ): Query | false | null {
 	return state?.route?.query?.current || null;
```

Comment:
I think the selector should just return whatever `state.route.query.current` points to. No need to coalesce `null` in the mix as we do here either (as also done previously). So either `undefined` or `state.route.query.current` -> ultimately, just `return state?.route?.query:?.current;`.